### PR TITLE
Rename `conflict` revset to `conflicts`.

### PR DESCRIPTION
**Pull request by essiene**
See discussion thread in linked issue.

With this PR, all revset functions in [BUILTIN_FUNCTION_MAP](https://github.com/martinvonz/jj/blob/8d166c764251042287626ab37a1148386cef8371/lib/src/revset.rs#L570)
that return multiple values are either named in plural or the naming is hard to misunderstand (e.g. `reachable`)

Fixes: #4122
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
I'm for this change, so I'll let someone else review this PR. (Also, Essien and I work for the same company, so there's also that potential conflict of interest.)

**Comment by martinvonz** [comment]
> I'm for this change, so I'll let someone else review this PR.

To clarify, I don't mean that only someone who's against this should review it :) I just mean that I've already said elsewhere that I'm for it, so I want to give others a chance to say what they think, in particular Yuya who seemed ambivalent.

**Review by thoughtpolice — APPROVED** [review]
I'm completely OK with this, though maybe an alias with a warning wouldn't hurt. (Someone else can hold this PR back if they think that's important enough.)

File: lib/src/revset.rs
Comment (yuja) [inline lib/src/revset.rs:776]: I think it's better to leave the original "conflict" function for a while (with TODO comment about deprecation warning.)

We'll probably need to rename `commit.conflict()` template function for consistency, and it's more likely to break user's stock templates.

I have no idea whether `--conflict`/`--conflicted` should be renamed to `--conflicts`, btw.

File: CHANGELOG.md
Comment (yuja) [inline CHANGELOG.md:1]: In commit message:
>  With this PR, all revset functions in [BUILTIN_FUNCTION_MAP](https://github.com/martinvonz/jj/blob/8d166c764251042287626ab37a114838>
> that return multiple values are either named in plural or named in a difficult-to-misused manner (e.g. `reachable`)

nit: this is not true. `file()` for example returns multiple revisions, but is not plural. I'm not saying it should be renamed to `files()`, but I mean `conflicts()` in this context can be read as "any revision has one or more conflicts."

File: lib/src/revset.rs
Comment (martinvonz) [reply lib/src/revset.rs:776]: > I think it's better to leave the original "conflict" function for a while (with TODO comment about deprecation warning.)

Yes, good point (you and @thoughtpolice), especially since we know there are some tools that depend on the CLI (was it lazyjj? jj_tui?).

Comment (essiene) [reply lib/src/revset.rs]: A good sould implemented diagnostics while I wasn't looking :-D
So I've gone ahead to add the deprecation warnings.

File: CHANGELOG.md
Comment (essiene) [reply CHANGELOG.md:1]: thanks for pointing that out.

I've actually gone ahead to also rename `file()` to `files()`, by the same logic, so it is consistent.

Comment (essiene) [reply CHANGELOG.md:1]: Thanks for checking and confirming. I'll go ahead to submit.

File: CHANGELOG.md
Comment (martinvonz) [inline CHANGELOG.md:100]: Should `file` have been mentioned here?

Comment (essiene) [reply CHANGELOG.md:100]: My bad. I forgot to come back to check this.

I've just sent #4529 to update the CHANGELOG.

File: CHANGELOG.md
Comment (yuja) [reply CHANGELOG.md:1]: Oh, I mean the rationale described in changelog and commit message doesn't make sense:
> With this, all revsets that could potentially return multiple results are named in the plural.

The change looks okay to me (though I don't think `files()` is better than `file()`.)

Comment (essiene) [reply CHANGELOG.md:1]: I completely misunderstood that. My bad.

**Pull request merged**
